app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartController.scala

```diff
@@ -88,17 +88,15 @@ class StartController @Inject()(
     }
   }
 
-  def returnAfterMapping(id: Option[String] = None): Action[AnyContent] = Action.async { implicit request =>
+  def returnAfterMapping(): Action[AnyContent] = Action.async { implicit request =>
     withSubscribingAgent { agent =>
+      val sjr = agent.getMandatorySubscriptionRecord
       continueUrlActions.withMaybeContinueUrlCached {
-        id match {
-          case Some(continueId) =>
-            for {
-              record <- subscriptionJourneyService.getMandatoryJourneyRecord(ContinueId(continueId))
-              _      <- subscriptionJourneyService.saveJourneyRecord(record.copy(mappingComplete = true))
-            } yield Redirect(routes.TaskListController.showTaskList())
-
-          case None => Future.successful(Redirect(routes.TaskListController.showTaskList()))
+        sjr.authProviderId flatMap { authProviderId =>
+          for {
+            record <- subscriptionJourneyService.getMandatoryJourneyRecord(authProviderId)
```

Comment:
Not sure why you can't update the sjr you got above?  Why do you need to refetch it again?

Reply:
would this work too:

```
def returnAfterMapping(): Action[AnyContent] = Action.async { implicit request =>
    withSubscribingAgent { agent =>
      continueUrlActions.withMaybeContinueUrlCached {
        val sjr = agent.getMandatorySubscriptionRecord
        subscriptionJourneyService.saveJourneyRecord(sjr.copy(mappingComplete = true))
          .map(_ => Redirect(routes.TaskListController.showTaskList()))
      }
    }
  }
```

Reply:
yeah sorry, too excited about lunch, will fix now